EFCoreAssignment/Program.cs

```diff
@@ -20,30 +21,153 @@
 LoadingRelatedData_ExplicitLoading(dbContext);
 LoadingRelatedData_EagerLoading(dbContext);
 
+InsertProduct(dbContext);
+InsertProductWithNewShop(dbContext);
+UpdateProduct(dbContext);
+DeleteProduct(dbContext);
+DeleteProductByKey(dbContext);
+
+static void InsertProduct(AppDbContext dbContext)
+{
+    var product = new Product
+    {
+        Name = "McFlurry Oreo",
+        ShopId = 3
+    };
+
+    dbContext.Products.Add(product);
+    dbContext.SaveChanges();
+}
+
+static void InsertProductWithNewShop(AppDbContext dbContext)
+{
+    var shop = new Shop
+    {
+        Name = "Coco"
+    };
+
+    var product = new Product
+    {
+        Name = "Ice Water",
+        Shop = shop
+    };
+
+    dbContext.Shops.Add(shop);
```

Comment:
No need to add **shop** using **dbContext.Add(shop)**. Shop is a navigational property in Product Entity.

You can remove this line of code. Thank you

Reply:
Updated code. Thanks!